packages/landing/src/index.ts

```diff
@@ -4,24 +4,12 @@ import { Basemaps } from './map';
 import { addDebugLayer } from './debug';
 import { WindowUrl } from './url';
 import { BasemapsUi } from './ui';
+import { isWebpSupported } from './webp';
 
-// Source https://stackoverflow.com/questions/5573096/detecting-webp-support
-function canUseWebP(): boolean {
-    const elem = document.createElement('canvas');
-
-    if (!!(elem.getContext && elem.getContext('2d'))) {
-        // was able or not to get WebP representation
-        return elem.toDataURL('image/webp').indexOf('data:image/webp') == 0;
-    }
-
-    // very old browser like IE 8, canvas not supported
-    return false;
-}
-
-document.addEventListener('DOMContentLoaded', () => {
+document.addEventListener('DOMContentLoaded', async () => {
     const mapEl = document.getElementById('map');
     if (mapEl == null) throw new Error('Cannot find #map element');
-    if (canUseWebP()) WindowUrl.ImageFormat = 'webp';
+    if (await isWebpSupported) WindowUrl.ImageFormat = 'webp';
 
```

Comment:
Shouldn't this be a function call?

Reply:
Its currently just a Promise, as you dont need to wait for `DOMContentLoaded` to start testing, I was wondering if we should wait though...

so when you import the file, it tries to render a webp image, rendering looks to take 20+ms so we could add it to DOMContentLoaded event

Reply:
Ignore that